File: 브루트 포스/2231/2231.py
import math

def decompose(n):
   d = math.floor(math.log10(n) + 1)
   for i in range(n - d * 10, n):
      temp = j = i
      while j > 0:
         temp += j % 10
         j //= 10
      if n == temp:
         return i
   return 0

n = int(input())
print(decompose(n))

File: 브루트 포스/2231/2231.cpp
#include <iostream>
#include <cmath>
using namespace std;

int decompose(int);

int main()
{
   int n;
   cin>>n;
   cout<<decompose(n);

   return 0;
}

int decompose(int n)
{
   int d = floor(log10(n) + 1);
   for (int i = n - d * 10; i < n; ++i)
   {
      int temp = i;
      for (int j = i; j > 0; j /= 10)
         temp += j % 10;
      if (n == temp)
         return i;
   }
   return 0;
}